src/eventStore/dynamodb/index.ts

```diff
@@ -0,0 +1,132 @@
+    this.config = config;
+  }
+
+  async connect(): Promise<void> {
+    const config: any = {
+      region: this.config.region,
+    };
+
+    this.config.secretAccessKey &&
+      (config.secretAccessKey = this.config.secretAccessKey);
+    this.config.accessKeyId && (config.accessKeyId = this.config.accessKeyId);
+
+    dynamoose.aws.sdk.config.update(config);
+  }
+
+  async addEvent(
+    contractAddress: string,
+    abi: AbiItem,
+    {
+      blockNumber,
+      options,
+    }: { blockNumber?: number; options?: Record<string, unknown> } = {},
+  ): Promise<void> {
+    let contract = await AntennaModel.get({
+      contractAddress,
+    });
+    if (!contract) {
+      contract = await AntennaModel.create({
+        contractAddress,
+      });
+    }
+
+    const arr = contract.abi || [];
+    arr.push(abi);
+    contract.abi = arr;
+    if (blockNumber) {
+      contract.blockNumber = blockNumber;
+    }
+    if (options) {
+      contract.options = options;
+    }
+    await contract.save();
+  }
+
+  async updateEvent(
+    contractAddress: string,
+    abi: AbiItem[],
+    {
+      blockNumber,
+      options,
+    }: { blockNumber?: number; options?: Record<string, unknown> } = {},
+  ): Promise<void> {
+    let contract = await AntennaModel.get({
+      contractAddress,
+    });
+    if (!contract) {
+      contract = await AntennaModel.create({
+        contractAddress,
+      });
+    }
+
+    contract.abi = abi;
+    if (blockNumber) {
+      contract.blockNumber = blockNumber;
+    }
+    if (options) {
+      contract.options = options;
+    }
+    await contract.save();
+  }
+
+  async removeEvent(contractAddress: string, eventName: string): Promise<void> {
+    const contract = await AntennaModel.get({
+      contractAddress,
+    });
+    if (!contract) {
+      throw Error('invalid contract address');
+    }
+
+    const event = contract.abi.find((event) => event.name === eventName);
+    if (!event) {
+      throw Error('invalid event name');
+    }
+    contract.abi = contract.abi.splice(contract.abi.indexOf(event), 1);
+    if (contract.abi.length === 0) {
+      await contract.delete();
+    } else {
+      await contract.save();
+    }
+  }
+
+  async updateBlock(
+    contractAddress: string,
+    blockNumber: number,
+  ): Promise<void> {
+    await AntennaModel.update({ contractAddress }, { blockNumber });
+  }
+
+  async getContract(contractAddress: string): Promise<Contract> {
+    const contract = await AntennaModel.get({
+      contractAddress,
+    });
+    if (!contract) {
+      throw Error('invalid contract address');
+    }
+    return {
+      contractAddress: contract.contractAddress,
+      abi: contract.abi,
+      blockNumber: contract.blockNumber,
+      options: contract.options,
+    };
+  }
+
+  async getContracts(): Promise<Contract[]> {
+    const contracts = await AntennaModel.scan().all().exec();
+
+    return contracts.map((contract: any) => ({
```

Comment:
I know DynamoDB is schemaless DB. But it will be helpful if you describe contract type for auto completion and type safety.